config/core/resources/subscription.yaml

```diff
@@ -127,6 +127,7 @@ spec:
                       namespace:
                         description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/ This is optional field, it gets defaulted to the object holding it if left out.'
                         type: string
+                    x-kubernetes-preserve-unknown-fields: true # This is necessary to enable the experimental feature
```

Comment:
There has to be many many many more places `krefs` are used, like `broker.spec.config`, `trigger.spec.subscriber`, `source.spec.sink`, etc. Why not update those too?

Reply:
It's explained in the plan here: https://github.com/knative/eventing/issues/5086

In short, I want to be incremental and avoid a gigantic PR where I implement this everywhere, my idea is to merge initially only for subscriber.ref, then go ahead and implement it in the other places where we use it. As you see from the plan, I also state that a "requirement" for me for the feature to graduate to beta is to make sure that everywhere we use `KRef`, we should support this experimental feature.